File: languages/java/secure-memory/src/test/java/com/godaddy/asherah/securememory/protectedmemoryimpl/ProtectedMemoryExceptionTest.java
package com.godaddy.asherah.securememory.protectedmemoryimpl;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ProtectedMemoryExceptionTest {
  static String message = "Test Message";

  @Test
  void protectedMemoryExceptionConstructorTest() {
    ProtectedMemoryException ex = new ProtectedMemoryException(message);
    assertEquals(message, ex.getMessage());
  }
}

File: csharp/SecureMemory/SecureMemory.Tests/ProtectedMemoryImpl/ProtectedMemoryExceptionTest.cs
using GoDaddy.Asherah.SecureMemory.ProtectedMemoryImpl;
using Xunit;

namespace GoDaddy.Asherah.SecureMemory.Tests.ProtectedMemoryImpl
{
    [Collection("Logger Fixture collection")]
    public class ProtectedMemoryExceptionTest
    {
        private const string Message = "Failure message";

        [Fact]
        private void ProtectedMemoryExceptionConstructorTest()
        {
            SecureMemoryException exception = new SecureMemoryException(Message);
            Assert.Equal(Message, exception.Message);
        }
    }
}
